crossbeam-utils/src/timestamp.rs

```diff
@@ -0,0 +1,16 @@
+/// Trait representing basic operations on a timestamp value.
+/// Timestamps are often used in lock-free protocols to implement
+/// versioned values.
+pub trait Timestamp {
+    /// Create a new timestamp.
+    fn new() -> Self;
+
+    /// Check `self` is earlier than `other`.
+    fn earlier(&self, other: &Self) -> bool;
```

Comment:
Does it make sense to implement `PartialOrd` for timestamps instead of custom `earlier`/`later` methods?

Reply:
These don't observe transitivity.